packages/palette/src/elements/DonutChart/DonutChart.tsx

```diff
@@ -0,0 +1,164 @@
+import { interpolate } from "d3-interpolate"
+import { arc as d3_arc, pie as d3_pie } from "d3-shape"
+import React, { useRef, useState } from "react"
+import { Spring } from "react-spring"
+import styled from "styled-components"
+import { color, space } from "../../helpers"
+import { Color } from "../../Theme"
+import { Box } from "../Box"
+import { ChartHoverTooltip } from "../DataVis/ChartHoverTooltip"
+import { coerceTooltip } from "../DataVis/ChartTooltip"
+import { ProvideMousePosition } from "../DataVis/MousePositionContext"
+import { ChartProps } from "../DataVis/utils/SharedTypes"
+import { useGetWrapperWidth } from "../DataVis/utils/useGetWrapperWidth"
+import { useHasEnteredViewport } from "../DataVis/utils/useHasEnteredViewPort"
+import { Sans } from "../Typography"
+
+const colors: Color[] = ["black10", "black30", "black60"]
+
+export interface DonutChartProps extends ChartProps {
+  margin?: number
+}
+
+/**
+ * DonutChart is a component that displays data points with donut shaped arcs.
+ * Good for illustrating numerical proportions.
+ */
+export const DonutChart: React.FC<DonutChartProps> = ({
+  points,
+  margin = space(3),
+}: DonutChartProps) => {
+  const [hoverIndex, setHoverIndex] = useState(-1)
+  const [rest, setRest] = useState(false)
+
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  const hasEnteredViewport = useHasEnteredViewport(wrapperRef)
+
+  const width = useGetWrapperWidth(wrapperRef)
+
+  const donutWidth = width * 0.12
+  const donutLabelMargin = space(2) / Math.sqrt(2)
+
+  const centerX = width / 2 + margin
+  const centerY = width / 2 + margin
+
+  const values = points.map(d => d.value).sort()
+  const angelInterpolator = interpolate(0, 2 * Math.PI)
+
+  const arc = d3_arc()
+    .innerRadius(width / 2 - margin - donutWidth)
+    .outerRadius(width / 2 - margin)
+    .padAngle(Math.PI / 360)
+
+  // virtual donut to use its arc centroids to calculate label position
+  const labelArc = d3_arc()
+    .innerRadius(width / 2 - margin + donutLabelMargin)
+    .outerRadius(width / 2 - margin + donutLabelMargin)
+    .padAngle(Math.PI / 360)
+
+  const pie = d3_pie()
+
+  const tooltip = points.map((point, index) => {
+    const hover = hoverIndex === index
+    return hover ? (
+      <ChartHoverTooltip key={`${index}-hover`}>
+        {coerceTooltip(point.tooltip)}
+      </ChartHoverTooltip>
+    ) : null
+  })
+
+  const labels = pie(values).map((slice, index) => {
+    const [x, y] = labelArc.centroid(slice as any)
+    const axisLabelX = points[index].axisLabelX
+    return (
+      axisLabelX && (
+        <DonutLabelContainer
+          key={index}
+          opacity={rest ? 1 : 0}
```

Comment:
These labels fade in a bit too late IMO. Feels sluggish. Would be good if they were already appearing just as the donut animation is ending.

Reply:
I agree. I can trigger the opacity when `num` is greater than, let's say 0.7 instead of waiting for the spring animation to end.

Reply:
Yeah made that change looks much better ❤️